### PR TITLE
Allow setting ACL through py-bsd

### PR DESCRIPTION
**Pull request by anodos325**
Original version was ultimately relying on output of acl_from_text() to set the ACL.
We want to be able to set an ACL without this information. Also removed the 'name' field.

### CODE REVIEW

File: bsd/acl.pyx [hunk 1]
```diff
@@ -259,33 +259,30 @@ cdef class ACLEntry(object):
...
-            'name': self.name,
             'type': self.type.name,
             'perms': {k.name: v for k, v in self.perms.items()},
             'flags': {k.name: v for k, v in self.flags.items()},
             'text': self.text
         }
 
     def __setstate__(self, obj):
-        if 'text' in obj:
+        if 'text' in obj and obj['text'] is not None:
```
Comment (skarekrow) [inline bsd/acl.pyx]:
```suggestion
        if 'text' in obj and obj.get('text'):
```
Is cleaner to me

File: bsd/acl.pyx [hunk 2]
```diff
@@ -259,33 +259,30 @@ cdef class ACLEntry(object):
...
             self.text = obj['text']
 
-        if 'id' in obj:
-            self.id = obj['id']
+        if 'tag' in obj:
+            self.tag = obj['tag']
 
-        if 'name' in obj:
-            self.name = obj['name']
+        if 'id' in obj and obj['id'] is not None:
```
Comment (skarekrow) [inline bsd/acl.pyx]: Same here

**Review by skarekrow — DISMISSED** [review]

File: bsd/acl.pyx (hunk 2 above)
Comment (anodos325) [reply bsd/acl.pyx]: We need an explicit check for none type here. You can have id 0. bsd.acl sets "id = None" if ae_tag is ACL_USER_OBJ, ACL_GROUP_OBJ, or ACL_EVERYONE.

File: bsd/acl.pyx (hunk 1 above)
Comment (anodos325) [reply bsd/acl.pyx]: I'll do one better and remove it entirely. I don't see a reason to have the text output. :)

File: bsd/acl.pyx (hunk 2 above)
Comment (sonicaj) [reply bsd/acl.pyx]: `obj.get('id') is not None`, we can drop the first if as well ; )

**Review by sonicaj — COMMENTED** [review]
@anodos325 has this been leftover intentionally - https://github.com/freenas/py-bsd/blob/e543f5cadbd185d65dbea4a4d574f53ff7b9351f/bsd/acl.pyx#L132 ? I am not sure, so confirming

**Comment by anodos325** [comment]
> @anodos325 has this been leftover intentionally -
> 
> [py-bsd/bsd/acl.pyx](https://github.com/freenas/py-bsd/blob/e543f5cadbd185d65dbea4a4d574f53ff7b9351f/bsd/acl.pyx#L132)
> 
> Line 132 in [e543f5c](/freenas/py-bsd/commit/e543f5cadbd185d65dbea4a4d574f53ff7b9351f)
>  self.text = text 
> ? I am not sure, so confirming

Cleaned that one up. Wrote new code so I probably introduced whitespace errors everywhere :)

File: bsd/acl.pyx [hunk 3]
```diff
@@ -259,53 +317,55 @@ cdef class ACLEntry(object):
...
 
-        if 'name' in obj:
-            self.name = obj['name']
-
         if 'type' in obj:
             self.type = obj['type']
 
         if 'perms' in obj:
-            for k, v in obj['perms']:
-                self.perms[k] = v
```
Comment (william-gr) [inline bsd/acl.pyx:284]: I think the reason it was using a for loop was to copy the dict.
Just setting the attribute you're changing the behavior if you change the dict that was passed to bsd.acl as argument.

Comment (william-gr) [reply bsd/acl.pyx:284]: tl;dr: dict is not a singleton

**Review by william-gr — APPROVED** [review]

**Pull request merged**